### ISSUE TITLE
try except block ignored when raising reportOptionalMemberAccess

### ISSUE THREAD

**Issue by jontwo** [microsoft/pyright#10508]
## Environment data

-   Pylance version: 2025.5.1
-   OS and version: Ubuntu 22.04.5 LTS
-   Python version (& distribution if applicable, e.g. Anaconda): 3.12.10

## Code Snippet

```python
    try:
        result = re.search(patt, txt).group(1)
    except AttributeError:
        # do something
        ...
```

## Repro Steps

Doing a simple try..except on a regex result gives the warning `"group" is not a known attribute of "None"`. I know that, that's why I'm using a try..except.

## Expected behavior

No warning should be given. Pylance should know that I am catching the `AttributeError`.

## Actual behavior

A warning appears in the Problems panel that should not be there.
